packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx

```diff
@@ -0,0 +1,104 @@
+import {Component, Prop, h, Listen} from '@stencil/core';
+import {buildInteractiveResult, Result, SmartSnippet} from '@coveo/headless';
+import {LinkWithResultAnalytics} from '../result-link/result-link';
+import {
+  Bindings,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../utils/initialization-utils';
+import {ResultContextEvent} from '../result-template-components/result-template-decorators';
+
+@Component({
+  tag: 'atomic-smart-snippet-source',
+  shadow: false,
+})
+export class AtomicSmartSnippetSource implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @Prop() source!: Result;
+  @Prop() smartSnippet: SmartSnippet | undefined;
+  @Prop() isSuggestion!: boolean;
```

Comment:
As discussed in [this thread](https://github.com/coveo/ui-kit/pull/1962/files#r865404099), I would recommend removing this prop only using the component in `smart-snippet` for now. I'll add it in `smart-snippet-suggestions` when the headless controller is ready.